.github/workflows/forest.yml

```diff
@@ -74,6 +74,28 @@ jobs:
             ~/.cargo/bin/forest*
           if-no-files-found: error
 
+  cargo-publish-dry-run:
+    name: cargo publish --dry-run
+    runs-on: ubuntu-latest
+    steps:
+      # To help investigate transient test failures
+      - run: lscpu
+      # find the nearest S3 space for storing cache files
+      - name: Show IP
+        run: curl ifconfig.me
+        continue-on-error: true
```

Comment:
Why do we need it in this job exactly? If there isn't anything special about it, let's either have it everywhere (ideally via a composite action to reduce code duplication) or only in critical places.

Reply:
Removed